### ISSUE TITLE
Improve performance with `fixedrate=true`

### ISSUE THREAD

**Issue by heliosdrm** [JuliaDynamics/RecurrenceAnalysis.jl#47]
In the calculation of recurrence matrices with fixed recurrence rate there is the issue commented in #46 (indirectly calling `distancematrix`, which is expensive):

https://github.com/JuliaDynamics/RecurrenceAnalysis.jl/blob/2961a32b29e41ea1e3dc3d63f67e29d74948e5ab/src/matrices.jl#L220-L222

A function that gives the same result as `quantile(distancematrix(x,y))` without calling `distancematrix` would help.

**Comment by heliosdrm** [comment]
Hint: the fixed recurrence rate `ε` is expected to be a small number (order of magnitude 10^-2). I guess that this may help to find a fast method using only a small fraction of the memory taken by the full distance matrix.

**Comment by heliosdrm** [comment]
This might be a solution: https://discourse.julialang.org/t/package-for-approximate-small-quantiles/20046/2

Drawbacks: possibly heavy dependencies, instability (the Windows build of OnlineStats is persistently failing for 32-bits); in addition the calculation of the quantiles is not exact but approximate, and needs to set a parameter for the approximation, so the implementation is not straightforward.

So, although it is a solution that may deserve to be taken into account, it needs time to be analyzed. So we can leave it for version 2.0 if suitable (it would imply breaking changes in my opinion).

**Comment by Datseris** [comment]
Hm,,,, I must admit I find it difficult justify adding OnlineStats just for this. I'd much rather just document the possibility.

Our heaviest dependency so far is NearestNeighbors which only depends on StaticArrays. I have to say I am very happy with how low we have kept the dependencies on this package. I think this is a great benefit and I vote to keep it this way. 

I have been following some discussions in the Julia forums and some parties have made it clear that to install packages they have to pass through a screening of some sorts. And for that case each dependency matters significantly.

**Comment by pucicu** [comment]
I strongly agree! In particular for a starting user, each additional package that needs to be installed is a challenge for the patience of the user. He/She would just like to start quickly and not be bothered with installing again and again additional packages.
